src/components/Dialog.tsx

```diff
@@ -115,14 +126,82 @@ export const Modal = styled.div<{ color: Color }>`
   flex-direction: column;
   height: 100%;
   left: 0;
-  overflow: hidden;
   padding: 0.5em;
   position: absolute;
   right: 0;
   top: 0;
   z-index: ${Layer.DIALOG};
 `
 
+const slideInLeft = keyframes`
+  from {
+    transform: translateX(calc(100% - 0.25em));
+  }
+`
+const slideOutLeft = keyframes`
+  to {
+    transform: translateX(calc(0.25em - 100%));
+  }
+`
+const slideOutRight = keyframes`
+  to {
+    transform: translateX(calc(100% - 0.25em));
+  }
+`
+
+const HiddenWrapper = styled.div`
+  height: 100%;
+  left: 0;
+  overflow: hidden;
+  padding: 0.5em;
+  position: absolute;
+  top: 0;
+  width: 100%;
+
+  @supports (overflow: clip) {
+    overflow: clip;
+  }
+`
+
+export const DialogAnimationLengthMs = ms`250`
+
+const AnimationWrapper = styled.div`
+  ${Modal} {
+    animation: ${slideInLeft} ${DialogAnimationLengthMs}ms ease-in;
+
+    &.${Animation.PAGING} {
+      animation: ${slideOutLeft} ${DialogAnimationLengthMs}ms ease-in;
+    }
+    &.${Animation.CLOSING} {
+      animation: ${slideOutRight} ${DialogAnimationLengthMs}ms ease-out;
+    }
+  }
+`
+
+// Accounts for any animation lag
+const PopoverAnimationUpdateDelay = ms`100`
+
+/* Allows slide in animation to occur without popovers appearing at pre-animated location. */
+function AnimatedPopoverProvider({ children }: PropsWithChildren) {
+  const popoverRef = useRef<HTMLDivElement>(null)
+  const [updatePopover, setUpdatePopover] = useState(false)
+  useEffect(() => {
+    setTimeout(() => {
+      setUpdatePopover(true)
+    }, DialogAnimationLengthMs + PopoverAnimationUpdateDelay)
+  }, [])
+
+  return (
+    <PopoverBoundaryProvider value={{ boundary: popoverRef.current, updateTrigger: updatePopover }}>
```

Comment:
to clarify - with this, we are adding a 2nd PopoverBoundaryProvider to the widget, right? we might want to leave a comment here or with the main PopoverBoundaryProvider to tell future readers that Popovers in Dialogs are not subject to the same provider?

Reply:
There isn't really a main popover provider, since the provider in Swap doesn't wrap dialogs